integration/disk_flush_mutlins_test.go

```diff
@@ -0,0 +1,131 @@
+		ns1ROpts           = clROpts.SetBlockSize(ns1BlockSize)
+		ns2BlockSize       = 3 * time.Hour
+		ns2ROpts           = clROpts.SetBlockSize(ns2BlockSize)
+	)
+
+	ns1, err := namespace.NewMetadata(testNamespaces[0], namespace.NewOptions().SetRetentionOptions(ns1ROpts))
+	require.NoError(t, err)
+	ns2, err := namespace.NewMetadata(testNamespaces[1], namespace.NewOptions().SetRetentionOptions(ns2ROpts))
+	require.NoError(t, err)
+	opts := newTestOptions(t).SetTickInterval(tickInterval).SetNamespaces([]namespace.Metadata{ns1, ns2})
+
+	// Test setup
+	testSetup, err := newTestSetup(t, opts)
+	require.NoError(t, err)
+	defer testSetup.close()
+
+	clOpts := testSetup.storageOpts.CommitLogOptions()
+	testSetup.storageOpts = testSetup.storageOpts.SetCommitLogOptions(clOpts.SetRetentionOptions(clROpts))
+	filePathPrefix := clOpts.FilesystemOptions().FilePathPrefix()
+
+	// it's aligned to lcm of ns block sizes
+	now := testSetup.getNowFn()
+
+	// Start the server
+	log := testSetup.storageOpts.InstrumentOptions().Logger()
+	log.Info("disk flush multiple namespaces test")
+	require.NoError(t, testSetup.startServer())
+	log.Info("server is now up")
+
+	// Stop the server
+	defer func() {
+		require.NoError(t, testSetup.stopServer())
+		log.Info("server is now down")
+	}()
+
+	log.Info("generating test data")
+	// test data for ns1
+	ns1SeriesMaps := make(map[time.Time]generate.SeriesBlock)
+	ns1InputData := []generate.BlockConfig{
+		{[]string{"foo", "bar"}, 100, now},
+		{[]string{"foo", "baz"}, 50, now.Add(ns1BlockSize)},
+	}
+
+	// test data for ns2
+	ns2SeriesMaps := make(map[time.Time]generate.SeriesBlock)
+	ns2InputData := []generate.BlockConfig{
+		{[]string{"foo", "bar"}, 20, now},
+	}
+
+	for _, ns1Input := range ns1InputData {
+		// write the data for ns1, always
+		testSetup.setNowFn(ns1Input.Start)
+		testData := generate.Block(ns1Input)
+		ns1SeriesMaps[ns1Input.Start] = testData
+		require.NoError(t, testSetup.writeBatch(testNamespaces[0], testData))
+		log.Infof("wrote ns1 for time %v", ns1Input.Start)
+
+		// when applicable, write the data for ns2, too
+		for _, ns2Input := range ns2InputData {
+			if ns1Input.Start != ns2Input.Start {
+				continue
+			}
+			testData = generate.Block(ns2Input)
+			ns2SeriesMaps[ns2Input.Start] = testData
+			log.Infof("wrote ns2 for time %v", ns2Input.Start)
+			require.NoError(t, testSetup.writeBatch(testNamespaces[1], testData))
+		}
+	}
+	log.Infof("test data written successfully")
+
+	// Advance time to make sure all data are flushed. Because data
+	// are flushed to disk asynchronously, need to poll to check
+	// when data are written.
+	log.Infof("waiting until data is flushed")
+	testSetup.setNowFn(testSetup.getNowFn().Add(3 * ns1BlockSize))
+	require.NoError(t, waitUntilDataFlushed(filePathPrefix, testSetup.shardSet, testNamespaces[0], ns1SeriesMaps, tickInterval*12))
+	require.NoError(t, waitUntilDataFlushed(filePathPrefix, testSetup.shardSet, testNamespaces[1], ns2SeriesMaps, tickInterval*5))
```

Comment:
Timeouts seem pretty short here, i.e. 12 * 3 seconds and 5 * 3 seconds.  Perhaps just give them both a minute each or something?  Sometimes Travis gets huge interrupts, can't trust CI to be speedy.  If CI ever runs on a host with multi VMs that allows "bursting" we'll run into long pauses, etc from time to time.